test/src/folding/parsers/python_parser_test.dart

```diff
@@ -0,0 +1,251 @@
+            _FB(startLine: 17, endLine: 19, type: _T.singleLineComment),
+          ],
+        ),
+
+        _Example(
+          'Python. Pair characters in literals are ignored',
+          code: '''
+a = '{[(';
+b = ")]}";
+''',
+          expected: [],
+        ),
+
+        _Example(
+          'Python. Service comments do not form blocks',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        # [START section1]
+        # [END section1]
+        a = 5''',
+          expected: [
+            _FB(startLine: 0, endLine: 4, type: _T.indent),
+            _FB(startLine: 1, endLine: 4, type: _T.indent),
+          ],
+        ),
+
+        _Example(
+          'Python. One-liners expressions do not form blocks',
+          code: '''
+def printAgeType(age):                         # 0
+    type = "Minor" if age < 18 else "Adult"    # 1
+    print(type)                                # 2
+
+def squareNumbers(numbers):                    # 4
+    squaredNumbers = [x**2 for x in numbers]   # 5
+    print(squaredNumbers)                      # 6
+
+printAgeType(19)                               # 8
+squareNumbers([1, 2, 3, 4, 5])                 # 9''',
+          expected: [
+            _FB(startLine: 0, endLine: 2, type: _T.indent),
+            _FB(startLine: 4, endLine: 6, type: _T.indent),
+          ],
+        ),
+
+        _Example(
+          'Python. Return only highlight blocks if it does not contain indents',
+          code: '''
+numbers = [1, 
+    2, 
+    3, 
+    4, 
+    5
+]
+squaredNumbers = [x**2 for x in numbers]
+print(squaredNumbers)''',
+          expected: [_FB(startLine: 0, endLine: 5, type: _T.brackets)],
+        ),
+      ];
+
+      for (final example in examples) {
+        highlight.registerLanguage('language', python);
+        final highlighted = highlight.parse(example.code, language: 'language');
+
+        final sequences = SingleLineComments.byMode[python] ?? [];
+
+        final commentParser = SingleLineCommentParser.parseHighlighted(
+          text: example.code,
+          highlighted: highlighted,
+          singleLineCommentSequences: sequences,
+        );
+
+        final serviceComments = ServiceCommentFilter.filter(
+          commentParser.comments,
+          namedSectionParser: const BracketsStartEndNamedSectionParser(),
+        );
+
+        final codeLines = CodeLineBuilder.textToCodeLines(
+          text: example.code,
+          highlighted: highlighted,
+          commentsByLines: commentParser.getCommentsByLines(),
+        );
+
+        final pythonParser = PythonFoldableBlockParser().parse(
+          highlighted: highlighted,
+          serviceCommentsSources: serviceComments.map((e) => e.source).toSet(),
+          lines: codeLines,
+        );
+
+        expect(
+          pythonParser,
+          example.expected,
+          reason: '${example.name}, valid blocks',
+        );
+      }
+    });
+  });
+}
```

Comment:
Please also add tests for:
- Imports. Try all forms of import: `import a`, `from a.b import c`, `from a import b as c`, etc. See if there are more. Check if Python supports breaking lines in an import statement and test that if it does.
- Single line comment block at the beginning of the file. This is especially important as all Apache files have a license comment header.